src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -135,13 +148,41 @@ def on_time_step(self, event: Event):
 
         max_household_address_id = households["address_id"].max()
 
-        (pop, _, _,) = update_address_id_for_unit_and_sims(
+        (pop, self.households, _,) = update_address_id_for_unit_and_sims(
             pop,
-            moving_units=households,
+            moving_units=self.households,
             units_that_move_ids=movers["household_id"],
             starting_address_id=max_household_address_id + 1,
             unit_id_col_name="household_id",
             address_id_col_name="address_id",
         )
 
         self.population_view.update(pop)
+
+    ##################
+    # Helper methods #
+    ##################
+
+    def generate_initial_households(self, pop_data: SimulantData) -> pd.DataFrame():
+        households = self.population_view.subview(["household_id", "housing_type"]).get(
+            pop_data.index
+        )
+        households = (
+            households.drop_duplicates().sort_values("household_id").set_index("household_id")
+        )
+        households["address_id"] = np.arange(len(households))
+
+        return households
+
+    def generate_household_details(self, idx: pd.Index) -> pd.DataFrame:
+        pop = self.population_view.subview(["household_id"]).get(idx)
+        household_details = pop.join(
+            self.households,
+            on="household_id",
+        )
+        # Set housing_type dtypes
+        household_details["housing_type"] = household_details["housing_type"].astype(
+            pd.CategoricalDtype(categories=HOUSING_TYPES)
+        )
```

Comment:
Is the dtype of `self.households["housing_type"]` not Categorical already? Does the merge not preserve the dtype?

Reply:
It wasn't coming in as such